packages/ckeditor5-dev-env/tests/release-tools/utils/generatechangelogfromcommits-integration.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* jshint mocha:true */
+
+'use strict';
+
+const fs = require( 'fs' );
+const path = require( 'path' );
+const expect = require( 'chai' ).expect;
+const { tools } = require( '@ckeditor/ckeditor5-dev-utils' );
+const generateChangelogFromCommits = require( '../../../lib/release-tools/utils/generatechangelogfromcommits' );
+const { changelogHeader, getChangelog } = require( '../../../lib/release-tools/utils/changelog' );
+
+describe( 'dev-env/release-tools/utils', () => {
+	let tmpCwd, cwd;
+
+	describe( 'generateChangelogFromCommits() - integration test', () => {
+		before( () => {
+			cwd = process.cwd();
+			tmpCwd = fs.mkdtempSync( __dirname + path.sep );
+			process.chdir( tmpCwd );
+
+			exec( `git init` );
+
+			if ( process.env.CI ) {
+				exec( `git config user.email "ckeditor5@ckeditor.com"` );
+				exec( `git config user.name "CKEditor5 CI"` );
+			}
+
+			const packageJson = {
+				name: '@ckeditor/ckeditor5-test-package',
+				bugs: 'https://github.com/ckeditor/ckeditor5-test-package/issues',
+				repository: 'https://github.com/ckeditor/ckeditor5-test-package'
+			};
+
+			fs.writeFileSync(
+				path.join( tmpCwd, 'package.json' ),
+				JSON.stringify( packageJson, null, '\t' )
+			);
+		} );
+
+		after( () => {
+			exec( `rm -rf ${ tmpCwd }` );
+			process.chdir( cwd );
+		} );
+
+		it( 'generates a changelog for the first time', () => {
+			exec( 'git commit --allow-empty --message "Internal: An initial commit."' );
+
+			return generateChangelog( '0.0.1' )
+				.then( () => {
+					const changelog = getChangelog();
+
+					expect( changelog ).to.contain( changelogHeader );
+					expect( changelog ).to.contain( 'Internal changes only (updated dependencies, documentation, etc.).' );
+
+					release( '0.0.1' );
```

Comment:
Writing tests in this way is very risky. I'd expect that every single test cleans up after itself and every new test starts from point 0. Here, all the tests create a chain of releases. It makes some sense, but it's not a common practice and it may be confusing. You should perhaps write a comment at the beginning of this file which explains how these tests work.